src/application.c

```diff
@@ -43,6 +44,19 @@ static message_t build_status_message()
     msg.min_voltage = isnan(voltage_min) ? UINT8_MAX : (uint8_t)round(voltage_min * 10.0);
     msg.rssi = lora.rssi;
     msg.snr = lora.snr;
+
+    // Flag set logic.
+    
+    //msg.flags = 0; // Reset (DEBUG LINE) 
+    
+    if (last_uplink_confirmed) {
+        msg.flags |= FLAG_UPLINK_CONFIRMED;
+    }
+    if (lora.ack_received) {
+        msg.flags |= FLAG_ACK_RECEIVED;
+    }
```

Comment:
Do we need to reset the flag if no ack was received, i.e., do msg.flags &= ~FLAG_ACK_RECEIVED?

Reply:
Similarly, Yes, I wrote this before I realized the message is static and forgot to add that.
Have added this, will commit when other comments are resolved